taskcat/_cli_modules/test.py

```diff
@@ -31,6 +32,132 @@ class Test:
     Performs functional tests on CloudFormation templates.
     """
 
+    def __init__(
+        self,
+        test_names: str = "ALL",
+        regions: str = "ALL",
+        input_file: str = "./.taskcat.yml",
+        project_root: str = "./",
+        enable_sig_v2: bool = False,
+        minimal_output: bool = False,
+    ):
+        """Manages the lifecycle of stacks for a given taskcat test.
+
+        :param test_names: comma separated list of tests to run
+        :param regions: comma separated list of regions to test in
+        :param input_file: path to either a taskat project config file or a
+        CloudFormation template
+        :param project_root: root path of the project relative to input_file
+        :param enable_sig_v2: enable legacy sigv2 requests for auto-created buckets
+        :param minimal_output: Reduces output during test runs
+        """
+
+        self.uid = uuid.uuid4()
+        self.test_names = test_names
+        self.regions = regions
+
+        # Temporary attributes that need removed
+        self.minimal_output = minimal_output
+
+        project_root_path: Path = Path(project_root).expanduser().resolve()
+        input_file_path: Path = project_root_path / input_file
+
+        args = _build_args(enable_sig_v2, regions, GLOBAL_ARGS.profile)
+
+        self.config = Config.create(
+            project_root=project_root_path,
+            project_config_path=input_file_path,
+            args=args
+            # TODO: detect if input file is taskcat config or CloudFormation template
```

Comment:
I think for non-cli testing this is a pretty must-have feature. I think the best approach is to keep the current config file structure and behavior but instead create a class method for `Config` that will take a dictionary config file and return a `Config` object.

I might create that apart of this PR if it's not available.